packages/cli/generators/controller/index.js

```diff
@@ -34,25 +65,150 @@ module.exports = class ControllerGenerator extends ArtifactGenerator {
     return super.promptArtifactName();
   }
 
+  promptArtifactType() {
+    debug('Prompting for controller type');
+    return this.prompt([
+      {
+        type: 'list',
+        name: 'controllerType',
+        message: 'What kind of controller would you like to generate?',
+        when: this.artifactInfo.controllerType === undefined,
+        choices: [ControllerGenerator.BASIC, ControllerGenerator.CRUD],
+        default: ControllerGenerator.BASIC,
+      },
+    ])
+      .then(props => {
+        Object.assign(this.artifactInfo, props);
+        return props;
+      })
+      .catch(err => {
+        debug(`Error during controller type prompt: ${err.stack}`);
+        return this.exit(err);
+      });
+  }
+
+  promptArtifactCrudVars() {
+    let modelList = [];
+    let repositoryList = [];
+    if (
+      !this.artifactInfo.controllerType ||
+      this.artifactInfo.controllerType === ControllerGenerator.BASIC
+    ) {
+      return;
+    }
+    return utils
+      .getArtifactList(this.artifactInfo.modelDir, 'model')
+      .then(list => {
```

Comment:
We need to add `catch` here so that errors from `getArtifactList` will be caught for graceful exit. For example, if there is no `models` directory, running the generator throws the following error:
```
events.js:183
      throw er; // Unhandled 'error' event
      ^

Error: ENOENT: no such file or directory, scandir '/Users/rfeng/Projects/demos/x2/src/models'
```

Reply:
I've added a .catch at the end of the chain to call `this.exit` (it won't mark this comment as outdated)

Reply:
Unfortunately it does not work as your added `catch` only applies to `return this.prompt([`. I still see the same issue with your changes.